mlir/lib/Dialect/Arith/IR/InferIntRangeInterfaceImpls.cpp

```diff
@@ -42,13 +42,19 @@ void arith::ConstantOp::inferResultRanges(ArrayRef<ConstantIntRanges> argRanges,
   }
   if (auto arrayCstAttr =
           llvm::dyn_cast_or_null<DenseIntElementsAttr>(getValue())) {
+    assert(arrayCstAttr.size() && "Zero-sized vectors are not allowed");
+    if (arrayCstAttr.isSplat()) {
+      setResultRange(getResult(), ConstantIntRanges::constant(
```

Comment:
you can and an `if` before this one and check using `dyn_cast_or_null<SplatElementsAttr>(...)`

Reply:
Nice suggestion, done!